### PR TITLE
Feature/nsa 4893 prevent an active service request being re requested

### PR DESCRIPTION
**Pull request by stottidfe**
making sure that the end user cannot raise multiple service requests for the same service

### CODE REVIEW

File: src/app/home/views/services.ejs
Comment (ipredoi) [inline src/app/home/views/services.ejs:1]: The changes required for this ticket are very hard to follow in this file. Can we make sure we don't change anything else?

File: src/app/accessRequests/rejectSubServiceRequest.js [hunk 1]
```diff
@@ -1,78 +1,82 @@
...
-     // req.session.roleIds = undefined;
+  if (req.session.roleIds !== undefined) {
+    if (req.session.roleIds !== viewModel.role_ids) {
+      viewModel.role_ids = req.session.roleIds;
+      let submodel = await getRoleAndServiceNames(viewModel, req.params.rid, req);
+      viewModel.roles = submodel.roles.filter((x) => x !== undefined);
+      //req.session.roleIds = undefined;
+      req.session.roles = viewModel.roles;
+    } else {
+      // req.session.roleIds = undefined;
```
Comment (ipredoi) [inline src/app/accessRequests/rejectSubServiceRequest.js:26]: can we please remove the commented out lines?

File: src/app/accessRequests/rejectSubServiceRequest.js [hunk 2]
```diff
@@ -90,24 +94,25 @@ const post = async (req, res) => {
...
     return res.redirect(`/access-requests/requests`);
-  }
-  else
-  {
-      //reqId, statusId, approverId, reason
-    const request = await updateServiceRequest(req.params.rid,-1,req.user.sub ,model.reason);
-    if(request.success)
-    {
+  } else {
+    //reqId, statusId, approverId, reason
```
Comment (ipredoi) [inline src/app/accessRequests/rejectSubServiceRequest.js:113]: can we please remove the unused lines?

File: src/app/accessRequests/reviewSubServiceRequest.js [hunk 3]
```diff
@@ -47,32 +50,34 @@ const validate = async (req) => {
...
-     //req.session.roleIds = undefined;
-     req.session.roles = viewModel.roles;
-    }else{
-     // req.session.roleIds = undefined;
+  if (req.session.roleIds !== undefined) {
+    if (req.session.roleIds !== viewModel.role_ids) {
+      viewModel.role_ids = req.session.roleIds;
+      let submodel = await getRoleAndServiceNames(viewModel, req.params.rid, req);
+      viewModel.roles = submodel.roles.filter((x) => x !== undefined);
+      //req.session.roleIds = undefined;
```
Comment (ipredoi) [inline src/app/accessRequests/reviewSubServiceRequest.js:61]: can we please remove the unused lines?

File: src/app/organisations/organisations.js [hunk 4]
```diff
@@ -95,4 +95,4 @@ const organisations = async (req, res) => {
   });
 };
 
-module.exports = organisations;
+module.exports = {organisations, getAndMapPendingRequests};
```
Comment (ipredoi) [inline src/app/organisations/organisations.js:98]: can we remove getAndMapPendingRequests?

File: src/app/accessRequests/rejectSubServiceRequest.js (hunk 1 above)
Comment (stottidfe) [reply src/app/accessRequests/rejectSubServiceRequest.js]: done
